streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -281,48 +275,63 @@
+     * @param keySerde          key serdes for materializing the aggregated table,
+     *                          if not specified the default serdes defined in the configs will be used
+     * @param valueSerde        value serdes for materializing the aggregated table,
+     *                          if not specified the default serdes defined in the configs will be used
      */
     <W extends Window> KTable<Windowed<K>, V> reduceByKey(Reducer<V> reducer,
                                                           Windows<W> windows,
                                                           Serde<K> keySerde,
-                                                          Serde<V> aggValueSerde);
+                                                          Serde<V> valueSerde);
 
     /**
-     * Aggregate values of this stream by key on a window basis.
+     * Combine values of this stream by key on a window basis into a new instance of windowed {@link KTable}
+     * with default serializers and deserializers.
      *
-     * @param reducer the class of {@link Reducer}
+     * @param reducer the instance of {@link Reducer}
      * @param windows the specification of the aggregation {@link Windows}
      */
     <W extends Window> KTable<Windowed<K>, V> reduceByKey(Reducer<V> reducer, Windows<W> windows);
 
     /**
-     * Aggregate values of this stream by key on a window basis.
+     * Combine values of this stream by key into a new instance of ever-updating {@link KTable}.
      *
-     * @param reducer the class of Reducer
+     * @param reducer           the instance of {@link Reducer}
+     * @param keySerde          key serdes for materializing the aggregated table,
+     *                          if not specified the default serdes defined in the configs will be used
+     * @param valueSerde        value serdes for materializing the aggregated table,
+     *                          if not specified the default serdes defined in the configs will be used
+     * @param name              the name of the resulted {@link KTable}
      */
     KTable<K, V> reduceByKey(Reducer<V> reducer,
                              Serde<K> keySerde,
-                             Serde<V> aggValueSerde,
+                             Serde<V> valueSerde,
                              String name);
 
     /**
-     * Aggregate values of this stream by key on a window basis.
+     * Combine values of this stream by key into a new instance of ever-updating {@link KTable} with default serializers and deserializers.
      *
-     * @param reducer the class of {@link Reducer}
+     * @param reducer the instance of {@link Reducer}
+     * @param name    the name of the resulted {@link KTable}
      */
     KTable<K, V> reduceByKey(Reducer<V> reducer, String name);
 
     /**
-     * Aggregate values of this stream by key on a window basis.
+     * Aggregate values of this stream by key on a window basis into a new instance of {@link KTable}.
```

Comment:
To be consistent, we should say "into a new instance of [a windowed] {@link KTable}".